### PR TITLE
improving .assert_package()

### PR DESCRIPTION
**Pull request by larmarange**
new function `.get_package_dependencies()`

`.assert_package()` now checks the type of comparison for minimum version

closes https://github.com/ddsjoberg/gtsummary/issues/1296

### CODE REVIEW

**Comment by larmarange** [comment]
New function .get_package_dependencies() returning a tibble with all depencies, minimum version and comparison operator.

`.get_min_version_required()` now returns the comparison operator as an attribute

The comparison operator is taken into account by `.asset_package()`

**Comment by ddsjoberg** [comment]
Wow, so so fast! Thanks for adding this! I can review tonight!

**Comment by ddsjoberg** [comment]
Can we re-add that unit test we excluded on GH Actions? Does it pass now?

**Comment by larmarange** [comment]
> Can we re-add that unit test we excluded on GH Actions? Does it pass now?

I have updated the unit test. Let see if it is OK now

**Comment by ddsjoberg** [comment]
@larmarange everything looks perfect!

The only think I can think of is when multiples of the same package are installed (like below)

```
d> devtools::load_all(".")
ℹ Loading broom.helpers
d> .libPaths()
[1] "C:/Users/SjobergD/R-dev"            "C:/Program Files/R/R-4.2.1/library"
d> installed.packages() %>%
+   dplyr::as_tibble() %>%
+   dplyr::filter(Package == "broom.helpers") %>%
+   dplyr::select(Package, LibPath, Version)
# A tibble: 2 × 3
  Package       LibPath                            Version   
  <chr>         <chr>                              <chr>     
1 broom.helpers C:/Users/SjobergD/R-dev            1.7.0.9003
2 broom.helpers C:/Program Files/R/R-4.2.1/library 1.8.0     
d> .get_package_dependencies("gtsummary") %>%
+   dplyr::filter(pkg == "broom.helpers")
# A tibble: 2 × 6
  pkg_search pkg_search_version dependency_type pkg           version compare
  <chr>      <chr>              <chr>           <chr>         <chr>   <chr>  
1 gtsummary  1.6.1              Imports         broom.helpers 1.7.0   >=     
2 gtsummary  1.6.1              Imports         broom.helpers 1.7.0   >=     
```

There is a line that returns the most recent version (arranging by version number). But I think that the line that is picked should be the line associated with the first location listed in `.libPaths()`, since that is the version of the package that will be used.

For example, in the script above, I have two versions of broom.helpers installed in two locations. We should return the dependency information associated with the package saved in `"C:/Users/SjobergD/R-dev"` because that is the repo location that is listed first and hence the version of broom.helpers that will be used.

**Comment by larmarange** [comment]
> There is a line that returns the most recent version (arranging by version number). But I think that the line that is picked should be the line associated with the first location listed in `.libPaths()`, since that is the version of the package that will be used.

`installed.packages()` return a list ordered according to `.libPaths()`. If `remove_duplicates = T`, now this is the first one which is returned.

File: R/assert_package.R [hunk 1]
```diff
@@ -26,40 +37,70 @@ NULL
...
+      c("Package", "Version", "LibPath", "Imports", "Depends", "Suggests", "Enhances", "LinkingTo")
+    )) %>%
+    dplyr::rename(
+      pkg_search = "Package",
+      pkg_search_version = "Version",
+      lib_path = "LibPath"
+    )
+
+  if (!is.null(pkg_search))
+    deps <- deps %>% dplyr::filter(.data$Package %in% .env$pkg_search)
```
Comment (ddsjoberg) [inline R/assert_package.R:76]: Should this be `.data$pkg_search%in% .env$pkg_search`?

Comment (larmarange) [reply R/assert_package.R]: You are right!!! The col was renamed just before and I have done the change too quickly.
Solved!

**Pull request merged**